### ISSUE TITLE
Analyzer doesn't warn about old-style typedefs with implicit parameter types

### ISSUE THREAD

**Issue by leiboldm** [dart-lang/linter#2541]
If you forget to name the parameter when typedef-ing a function (an easy mistake to make if you're coming from c++), type checking will not work for the function parameters.

Example where analyzer shows no errors:

```
typedef R FunctionType<R, P>(P);

void main() {
  FunctionType<String, int> a = (int input) => (input % 2 == 0) ? "even" : "odd";
  print(a(23.3));
}
```

However, if you name the parameter, the analyzer does find type mismatches:

```
typedef R FunctionType<R, P>(P input);

void main() {
  FunctionType<String, int> a = (int input) => (input % 2 == 0) ? "even" : "odd";
  print(a(23.3));
}
```

shows: 5: The argument type 'double' cannot be assigned to the parameter type 'int'. [ARGUMENT_TYPE_NOT_ASSIGNABLE]
